### PR TITLE
Update RSpec configuration and details

### PR DESCRIPTION
**Pull request by christiannelson**
Problem
=======
Raygun Rails includes configuration for RSpec that is based on a fairly old version of the defaults.

Solution
========
I generated a new set of default configuration and weaved our enhancements in to the new configuration.

* Require rspec configuration files from spec/support.*
* Turn on aggregate failures for non-system specs
* Run non-system specs first, then system specs

### CODE REVIEW

**Review by eskfung — APPROVED** [review]

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -123,4 +115,10 @@
   # test failures related to randomization by passing the same `--seed` value
   # as the one that triggered the failure.
   Kernel.srand config.seed
+
+  # Run non-system specs before system specs. `--seed` still applies to ordering both sets.
+  config.register_ordering(:global) do |items|
+    systems, others = items.partition { |g| g.metadata[:type] == :system }
+    others.shuffle + systems.shuffle
```
[the remark this answers is not in the document]
Comment (christiannelson) [reply spec/spec_helper.rb]: I know right!? Got the inspiration from this a while ago: 

https://www.rubydoc.info/github/rspec/rspec-core/RSpec%2FCore%2FConfiguration:register_ordering

**Comment by christiannelson** [comment]
@mattbrictson No obligation, but if you have 2 minutes to give this a once over I'll hold off on merging.

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -85,11 +87,7 @@
   #   - http://rspec.info/blog/2012/06/rspecs-new-expectation-syntax/
   #   - http://www.teaisaweso.me/blog/2013/05/27/rspecs-new-message-expectation-syntax/
   #   - http://rspec.info/blog/2014/05/notable-changes-in-rspec-3/#zero-monkey-patching-mode
-  # config.disable_monkey_patching!
-
-  # This setting enables warnings. It's recommended, but in some cases may
-  # be too noisy due to issues in dependencies.
-  # config.warnings = true
+  config.disable_monkey_patching!
```
Comment (mattbrictson) [inline spec/spec_helper.rb:90]: Is why we have to do `RSpec.describe` instead of just `describe`?

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -123,4 +115,10 @@
   # test failures related to randomization by passing the same `--seed` value
   # as the one that triggered the failure.
   Kernel.srand config.seed
+
+  # Run non-system specs before system specs. `--seed` still applies to ordering both sets.
+  config.register_ordering(:global) do |items|
+    systems, others = items.partition { |g| g.metadata[:type] == :system }
+    others + systems
+  end
```
Comment (mattbrictson) [inline spec/spec_helper.rb:123]: Nice!

**Review by mattbrictson — APPROVED** [review]
Looks good! I'd like to steal this for my current project. 😄

File: spec/spec_helper.rb (hunk 2 above)
Comment (christiannelson) [reply spec/spec_helper.rb:90]: I think so. I personally like the terse version stylistically, but then thought monkey patching is out of fashion.

What do you prefer?

Comment (mattbrictson) [reply spec/spec_helper.rb:90]: I am fine with `RSpec.describe`.

**Pull request merged**